src/khoj/processor/conversation/utils.py

```diff
@@ -165,9 +167,14 @@ def generate_chatml_messages_with_context(
         rest_backnforths += reciprocal_conversation_to_chatml([user_msg, assistant_msg])[::-1]
 
     # Format user and system messages to chatml format
+    def return_structured_message(message, image_url):
+        if image_url and vision_enabled:
+            return [{"type": "text", "text": message}, {"type": "image_url", "image_url": {"url": image_url}}]
```

Comment:
Did this ultimately end up working as expected?

Reply:
yup!